tests/plugins/targetmanagers/readmeta/readmeta.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package readmeta
+
+import (
+	"time"
+
+	"github.com/facebookincubator/contest/pkg/target"
+	"github.com/facebookincubator/contest/pkg/types"
+	"github.com/facebookincubator/contest/pkg/xcontext"
+)
+
+// Name defined the name of the plugin
+var (
+	Name = "readmeta"
+)
+
+// Readmeta implements the contest.TargetManager interface.
+type Readmeta struct {
+}
+
+// ValidateAcquireParameters valides parameters that will be passed to Acquire.
+func (r Readmeta) ValidateAcquireParameters(params []byte) (interface{}, error) {
+	return nil, nil
+}
+
+// ValidateReleaseParameters valides parameters that will be passed to Release.
+func (r Readmeta) ValidateReleaseParameters(params []byte) (interface{}, error) {
+	return nil, nil
+}
+
+// Acquire implements contest.TargetManager.Acquire
+func (t *Readmeta) Acquire(ctx xcontext.Context, jobID types.JobID, jobTargetManagerAcquireTimeout time.Duration, parameters interface{}, tl target.Locker) ([]*target.Target, error) {
+	// test metadata exists
+	jobID2, ok1 := types.JobIDFromContext(ctx)
+	// note this must use panic to abort the test run, as this is a test for the job runner which ignore the actual outcome
```

Comment:
you can emit an event with job id and run id and verify by fetching them afterwards (there's a helper for that)

Reply:
Are you sure I can emit events from the target manager and/or reporter?
(test step works of course)

Reply:
Done in the test step, I think this is the only place that can easily do it